unidiff/patch.py

```diff
@@ -350,7 +350,7 @@ def _parse(self, diff, encoding):
     def from_filename(cls, filename, encoding=DEFAULT_ENCODING, errors=None):
         """Return a PatchSet instance given a diff filename."""
         with open_file(filename, 'r', encoding=encoding, errors=errors) as f:
-            instance = cls(f)
+            instance = cls(f, encoding)
```

Comment:
This shouldn't be necessary since the file is already opened using the given encoding, so from that moment on when reading from `f` we would be getting (already decoded) unicode strings.